src/app/icd/ICDNotifier.h

```diff
@@ -0,0 +1,97 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app/AppConfig.h>
+#include <lib/core/CHIPError.h>
+
+class ICDSubscriber;
```

Comment:
I think the problem with "Subscriber" is that it's too easy to get confused with Matter subscriptions, esp because of how ICD-related those are.

If ICDListener is not ok, how about ICDActivityObserver?